examples/basic_usage.jl

```
@@ -40,99 +42,109 @@ end
+                combined_df = reduce(vcat, all_dataframes)
+                println("\n📊 Combined DataFrame info:")
+                println("   - Total rows: $(nrow(combined_df))")
+                println("   - Total columns: $(ncol(combined_df))")
+                println("   - Total batches: $batch_count")
+            else
+                println("\n📊 No data found in table")
+            end
 
-        for arrow_table in selected_iterator
-            df = DataFrame(arrow_table)
-            push!(selected_dataframes, df)
+            # Test with specific columns
+            println("\nTesting column selection...")
+            if !isempty(all_dataframes) && !isempty(names(all_dataframes[1]))
+                selected_columns = names(all_dataframes[1])[1:min(2, length(names(all_dataframes[1])))]
+                println("Selecting columns: $selected_columns")
+
+                selected_iterator = RustyIceberg.read_table(table_path, metadata_path; columns=selected_columns)
+                selected_dataframes = DataFrame[]
+
+                for arrow_table in selected_iterator
+                    df = DataFrame(arrow_table)
+                    push!(selected_dataframes, df)
+                end
+
+                if !isempty(selected_dataframes)
+                    combined_selected = reduce(vcat, selected_dataframes)
+                    println("✅ Column selection successful!")
+                    println("   - Selected columns: $(names(combined_selected))")
+                    println("   - Rows: $(nrow(combined_selected))")
+                end
+            end
         end
 
-        if !isempty(selected_dataframes)
-            combined_selected = reduce(vcat, selected_dataframes)
-            println("✅ Column selection successful!")
-            println("   - Selected columns: $(names(combined_selected))")
-            println("   - Rows: $(nrow(combined_selected))")
+    catch e
+        println("❌ Error reading table: $e")
```

IMO we shouldn't swallow the exception by the end of this PR